blog/_posts/2021-05-04-effective-programming-scala-course.md

```diff
@@ -0,0 +1,98 @@
+at scale. It explains how to achieve typical programming tasks in Scala,
+such as modeling business domains, breaking down complex problems into simpler problems,
+manipulating data, or running parallel tasks. Throughout the course, you will also learn
+the best practices for writing high-quality Scala code that scales to large applications,
+how to handle errors, how to write tests, and how to leverage a productive development
+environment.
+
+The course features six auto-graded programming assignments, which have been designed to
+be both fun and close to real-world programs. Topics covered by the assignments include
+graphic animations, web applications, property-based testing, generic programming,
+and parallel data processing.
+
+Each week contains several video lectures of ~10 minutes each. Lectures are focused on
+a specific skill or concept, and they are generally followed by a quiz so that you
+can immediately assess your understanding of the lesson.
+
+The syllabus is organized as follows:
+
+- week 1: basics of the language, and how to model business
+  domains with sealed traits and case classes,
+- week 2: manipulate collections of data with the standard library,
+  and write business logic,
+- week 3: tooling around the Scala language (IDE, build tools,
+  debuggers), organize large programs into modules,
+- week 4: reason about code, write unit tests, property-based tests, and
+  integration tests in Scala,
+- week 5: contextual abstractions (aka “implicits” in Scala 2),
+- week 6: error handling, data validation, and concurrent programming.
+
+The course has been co-designed by myself (Julien Richard-Foy) and [Noel Welsh].
+Noel Welsh has a long experience in teaching Scala, both in the industry and
+within the [Scala Bridge] project.
+
+## Other Scala Courses
+
+The Scala Center and Martin Odersky have also been working hard to update all our
+existing courses to Scala 3 (except the courses that use Spark, which still use
+Scala 2.12).
+
+This is a huge effort, and some parts are still in progress. Here is the current
+status, at the time this blog is written:
+
+- The [Scala Specialization] is a suite of five courses that teach Scala and
+  data processing (possibly in parallel or distributed),
+  - [Functional Programming Principles in Scala] has been updated to Scala 3
+    (all the lectures and assignments have been updated),
+  - [Functional Program Design] has been updated to Scala 3 (all the lectures
+    and assignments have been updated),
+  - [Parallel Programming] has been updated to Scala 3 (only the assignments
+    have been updated),
+  - [Big Data Analysis with Scala and Spark] is still on Scala 2.12,
+  - [Functional Programming in Scala Capstone] is still on Scala 2.12,
+- [Programming Reactive Systems] has been updated to Scala 3 (only the
+  assignments have been updated).
+
+We will update our Spark courses to Scala 3 as soon as there is a stable release of
+Spark that supports Scala 3.
```

Comment:
```suggestion
We will update our Spark courses to Scala 3 as soon as there is a stable release of
Spark that supports Scala 3.
```
Perhaps mention that this may be because it supports 2.13 and that Scala 3 can use 2.13 libraries (that don't have macros)?